satellite/repair/repairer/segments.go

```diff
@@ -182,19 +182,20 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (s
 
 	// Download the segment using just the healthy pieces
 	segmentReader, failedPieces, err := repairer.ec.Get(ctx, getOrderLimits, getPrivateKey, redundancy, pointer.GetSegmentSize())
+	defer func() {
+		// Populate node IDs that failed piece hashes verification
+		var failedNodeIDs storj.NodeIDList
+		for _, piece := range failedPieces {
+			failedNodeIDs = append(failedNodeIDs, piece.NodeId)
+		}
```

Comment:
One concern I would have with this is that before the changes, the values of failedPieces are evaluated immediately.  If you defer, you'll be evaluating the values of failedPieces at the time of the return.  So if someone modifies failedPieces after the defer, you could get some unexpected results.  You could pass failedPieces into the deferred func.